internals/secrethub/credential_disable.go

```diff
@@ -0,0 +1,86 @@
+package secrethub
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+	"github.com/secrethub/secrethub-cli/internals/secrethub/command"
+	"github.com/secrethub/secrethub-go/internals/api"
+)
+
+// CredentialDisableCommand is a command that allows to disable an existing credential.
+type CredentialDisableCommand struct {
+	io          ui.IO
+	force       bool
+	fingerprint string
+	newClient   newClientFunc
+}
+
+// NewCredentialDisableCommand creates a new command for disabling credentials.
+func NewCredentialDisableCommand(io ui.IO, newClient newClientFunc) *CredentialDisableCommand {
+	return &CredentialDisableCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Register registers the command, arguments and flags on the provided Registerer.
+func (cmd *CredentialDisableCommand) Register(r command.Registerer) {
+	clause := r.Command("disable", "Disable a credential for usage on SecretHub.")
+
+	fingerprintHelp := fmt.Sprintf("Fingerprint of the credential to disable. At least the first %d characters must be entered.", api.ShortCredentialFingerprintMinimumLength)
+	clause.Arg("fingerprint", fingerprintHelp).StringVar(&cmd.fingerprint)
```

Comment:
Do we in the future want to be able to select a credential on something other than its fingerprint? (e.g. name, device name)

If we might want to do so, do we want to pass the fingerprint as an arg or a flag?

Reply:
Names are not unique. So it would not be possible to use that as a selector through a flag. We _could_ let a user select a credential from a list when we prompt for it. But that is compatible with the current implementation.